File: algorithms/easy/row_with_maximum_ones.ex
defmodule Solution do
  @spec row_and_maximum_ones(mat :: [[integer]]) :: [integer]
  def row_and_maximum_ones(mat) do
    mat
    |> Enum.with_index()
    |> Enum.reduce([0, 0], fn {row, i}, [r, ct] ->
      ones = Enum.count(row, &(&1 == 1))

      if ones > ct do
        [i, ones]
      else
        [r, ct]
      end
    end)
  end
end

File: algorithms/easy/row_with_maximum_ones.rs
impl Solution {
    pub fn row_and_maximum_ones(mat: Vec<Vec<i32>>) -> Vec<i32> {
        let mut ndx = 0;
        let mut ct = 0;
        for (i, row) in mat.into_iter().enumerate() {
            let ones = row.into_iter().filter(|n| *n == 1).count();
            if ones > ct {
                ndx = i;
                ct = ones;
            }
        }
        vec![ndx as i32, ct as i32]
    }
}
